10-vendor-dependencies/README.md

```diff
@@ -0,0 +1,189 @@
+# ZEP-10: Vendoring external dependencies
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+      - [Unit tests](#unit-tests)
+      - [e2e tests](#e2e-tests)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Summary
+
+Vendoring external dependencies involves copying third-party libraries code
+directly into the project under a `vendor/` directory. This document outlines
+the advantages and disadvantages of this approach, aiming to support the adoption
+of vendoring for the Zarf project.
+
+## Motivation
+
+Zarf is [advertised](https://docs.zarf.dev/) as:
+
+> a free and open-source tool that enables declarative creation & distribution
+> of software into air-gapped/constrained/standalone environments.
+
+Its primary objective is to facilitate software delivery to disconnected environments.
+Consequently, it is logical for Zarf itself to support building in such environments.
+
+### Goals
+
+- Enable hermetic (reproducible and consistent) builds.
+- Support builds in disconnected environments.
+- Provide tighter control over dependencies.
+
+### Non-Goals
+
+- Modify existing dependency management tool.
+
+## Proposal
+
+The Go community has long debated the merits of vendoring dependencies. Research
+reveals compelling arguments on both sides. This section explores these arguments
+and explains why vendoring was chosen for the Zarf project.
+
+#### Pros
+
+1. **Reproducible and Consistent Builds.**
+
+   Vendoring supports [hermetic builds](https://sre.google/sre-book/release-engineering/#hermetic-builds-nqslhnid),
+   ensuring consistent and predictable outcomes.
+
+2. **Self-contained Builds.**
+
+   Vendoring allows disconnected, self-contained builds immediately after cloning
+   the repository, with only the Go compiler required.
+
+3. **Faster CI/CD Builds Times.**
+
+  Local dependencies eliminate the need to fetch external dependencies, speeding
+  up builds.
```

Comment:
This is especially nice for Windows builds, where github actions caching doesn't work, so pulling all the Zarf dependencies for unit tests takes about 10 minutes